ethereum_derivable_account.md

```diff
@@ -0,0 +1,193 @@
+---
+aip: 122
+title: x-chain DAA authentication using Sign-in-With-Ethereum
+author: igor-aptos, brian, 0xmaayan, hardsetting
+Status: Draft
+type: Framework
+discussions-to (*optional): https://github.com/aptos-foundation/AIPs/issues/601
+created: 04/17/2025
+---
+
+# AIP-X - x-chain DAA authentication using Sign-in-With-Ethereum
+
+## Summary
+
+Implements authentication function giving Ethereum users native Aptos accounts,
+and allowing them to sign transaction on Aptos blockchain, directly from their Ethereum wallet (verified with Metamask).
+This is done using derivable account abstraction (AIP-113).
+
+## High-level Overview
+
+Allow Ethereum accounts (Ethereum secp256k1 private keys inside Metamask Wallets) to create Aptos transactions,
+and be able to autheneticate and execute them directly on the Aptos blockchain, without interaction with Ethereum Blockchain.
+We do so by utilizing ["Sign-in-with-Ethereum"](https://docs.metamask.io/wallet/how-to/sign-data/siwe/) specification, and defining
+a message format that gives enough details to users so they understand what they are signing reasonably well.
+
+## Impact
+
+This will allow users with Ethereum Wallets to be able to interact with Aptos blockchain
+
+## Alternative Solutions
+
+Alternatively, wallets from other chains can implement integration with Aptos natively, providing more seamless experience to the users.
+
+## Specification and Implementation Details
+
+dApp is required to issue Sign-in-with-Ethereum request to the wallet with:
+
+```
+const statement = `Please confirm you explicitly initiated this request from ${domain}. You are approving to execute transaction ${human_readable_aptos_entry_function} on Aptos blockchain (${network_name}).`
+
+const prefix = `${domain} wants you to sign in with your Ethereum account:\n${address}\n\n${statement}`
+const suffix = `URI: ${scheme}://${domain}\nVersion: 1\nChain ID: ${aptos_chain_id}\nNonce: ${aptos_txn_digest}\nIssued At: ${new Date().toISOString()}`
+
+ethereumWallet.signMessage(`${prefix}\n${suffix}`)
+```
+
+This will generate a signature of the following full message:
+
+```
+<domain> wants you to sign in with your Ethereum account:
+<ethereum_address>
+
+Please confirm you explicitly initiated this request from <domain>. You are approving to execute transaction <entry_function> on Aptos blockchain (<network_name>).
+
+URI: <scheme>://<domain>
+Version: 1
+Chain ID: <aptos_chain_id>
+Nonce: <aptos_txn_digest>
+Issued At: <date_iso_string>
+```
+
```

Comment:
this seems a bit repetitive, maybe just simplify with:


````suggestion
dApp is required to issue signMessage call with following message:

```
<domain> wants you to sign in with your Ethereum account:
<ethereum_address>

Please confirm you explicitly initiated this request from <domain>. You are approving to execute transaction <entry_function> on Aptos blockchain (<network_name>).

URI: <scheme>://<domain>
Version: 1
Chain ID: <aptos_chain_id>
Nonce: <aptos_txn_digest>
Issued At: <date_iso_string>
```
````

Reply:
done